### PR TITLE
Adding some explanation to README about how to cite HPX

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -129,7 +129,9 @@ Please find the project's gpg key, which is used to sign HPX releases
 Citing
 ======
 
-In publications the latest release of HPX can be cited as: |zenodo_doi|.
+In publications, please use our publication in The Journal of Open Source
```
Comment (hkaiser) [inline README.rst:132]:
```suggestion
In publications, please use our paper in The Journal of Open Source
```

**Review by msimberg — DISMISSED** [review]
Thanks @hkaiser! Did you mean to review your own PR? Or is it a question about whether we prefer `publication` or `paper`?

**Comment by hkaiser** [comment]
> Thanks @hkaiser! Did you mean to review your own PR? Or is it a question about whether we prefer `publication` or `paper`?

I noticed using the word 'publication' twice and I was too lazy to re-commit, so I went in and added a review comment here ;-)

**Review by diehlpk — APPROVED** [review]
LGTM!

**Comment by aurianer** [comment]
Thanks!

**Pull request merged**